index.d.ts

```diff
@@ -32,189 +171,526 @@ declare namespace Resource {
+
+      /** GUID of this object's sprite */
+      spriteId: string;
+
+      /** Indicates if this object is visible */
+      visible: boolean;
+  }
+
+  export interface Options {
+      /** Resource GUID */
+      id: string;
+
+      /** Internal resource type descriptor */
+      modelName: string;
+
+      /** Version string, unknown use */
+      mvc: string;
+
+      /** Resource name */
+      name: string;
+  }
+
+  export interface Room {
+      /** Resource name */
+      name: string;
+
+      /** Resource GUID */
+      id: string;
+
+      /** The name of the Room Creation code FP, relative to the Room folder itself. */
+      creationCodeFile: string;
+
+      /** Inherit Code from a Parent Room */
+      inheritCode: boolean;
+
+      /** Inherit Creation Order from a Parent Room */
+      inheritCreationOrder: boolean;
+
+      /** Inherit Layers from a Parent Room */
+      inheritLayers: boolean;
+
+      /** instanceCreationOrderIDs */
+      instanceCreationOrderIDs: Array<string>;
+      isDnD: boolean;
+
+      /** All your layers are placed here. */
+      layers: Array<Layer>;
+
+      /** Internal resource type descriptor */
+      modelName: string;
+
+      /** Parent Room ID */
+      parentID: string;
+
+      /** Physics setting of the room. */
+      physicsSettings: PhysicsSettings;
+
+      /** Version string, unknown use */
+      mvc: string;
+  }
+
+  export interface Layer {
+      __type: string;
+      name: string;
+      id: string;
+      depth: number;
+      grid_x: number;
+      grid_y: number;
+      hierarchyFrozen: boolean;
+      hierarchyVisible: boolean;
+      inheritLayerDepth: boolean;
+      inheritLayerSettings: boolean;
+      inheritSubLayers: boolean;
+      inheritVisibility: boolean;
+      instances: Array<Instance>;
+      layers: Array<Layer>;
+      m_parentID: string;
+      m_serialiseFrozen: boolean;
+      modelName: string;
+      mvc: string;
+      userdefined_depth: boolean;
+      visible: boolean;
+  }
+
+  export interface Instance {
+      name: string;
+      id: string;
+      colour: Colour;
+      creationCodeFile: string;
+      creationCodeType: string;
+      ignore: boolean;
+      inheritCode: boolean;
+      inheritItemSettings: boolean;
+      IsDnD: boolean;
+      m_originalParentID: string;
+      m_serialiseFrozen: boolean;
+      modelName: string;
+      name_with_no_file_rename: string;
+      objId: string;
+      properties: null;
+      rotation: number;
+      scaleX: number;
+      scaleY: number;
+      mvc: string;
+      x: number;
+      y: number;
+  }
+
+  export interface Colour {
+      Value: number;
+  }
+
+  export interface PhysicsSettings {
+      id: string;
+      inheritPhysicsSettings: boolean;
+      modelName: string;
+      PhysicsWorld: boolean;
+      PhysicsWorldGravityX: number;
+      PhysicsWorldGravityY: number;
+      PhysicsWorldPixToMeters: number;
+      mvc: string;
+  }
+
+  export interface roomSettings {
```

Comment:
Unless there's a specific reasoning behind it, interfaces should all be PascalCase